### ISSUE TITLE
Ruby releaser doesn't parse prerelease versions properly

### ISSUE THREAD

**Issue by coaxial** [google-github-actions/release-please-action#742]
### TL;DR

For Rubygems, the way to number the prerelease for v0.10.0 would be `0.10.0.pre.1` in the gem's `version.rb` file. `release-please` doesn't understand that version number however and will just try to release `0.10.0` instead. `0.10.0-pre1` works with `release-please`, but this isn't the Rubygems convention and the resulting `version.rb` file has the (incorrect) `0.10.0-pre1`.

### Expected behavior

I expected `release-please` to create a release for version `0.10.0.pre.1`.

### Observed behavior

`release-please` creates release `0.10.0`, discarding the rest of the version number.

### Action YAML

```yaml
name: Release

on:
  push:
    branches:
      - master

permissions:
  contents: write
  pull-requests: write

jobs:
  release-please:
    runs-on: ubuntu-latest
    steps:
      - name: Prepare release
        uses: google-github-actions/release-please-action@v3
        with:
          release-type: ruby
          package-name: maid
          # Major bump is minor instead pre-1.0.0
          bump-minor-pre-major: true
          # Minor bump is patch instead pre-1.0.0
          bump-patch-for-minor-pre-major: true
          # Don't parse commits before this one as they don't follow the
          # conventional commit format AND don't have `release-please`
          # COMMIT_MESSAGE_OVERRIDE data.
          # (https://www.conventionalcommits.org/en/v1.0.0/#summary)
          bootstrap-sha: 0a6ff8abc6db00aca181d6bfed9751e332900ac9
          # This is the last release before starting to use release-please
          last-release-sha: 0a6ff8abc6db00aca181d6bfed9751e332900ac9
          version-file: lib/maid/version.rb
          token: ${{ secrets.MAID_GH_TOKEN }}
          prerelease: true
          release-as: v0.10.0.pre.1
      - uses: actions/checkout@v3
      - name: Set up Ruby
        uses: ruby/setup-ruby@v1
        with:
          # ruby-version derived from .ruby-version file
          bundler-cache: true
      - name: Install dependencies
        run: bundle install
      - name: Run tests
        run: bundle exec rake
      - name: Publish gem
        run: |
          mkdir -p $HOME/.gem
          touch $HOME/.gem/credentials
          printf -- "---\n:rubygems_api_key: ${GEM_HOST_API_KEY}\n" > $HOME/.gem/credentials
        env:
          GEM_HOST_API_KEY: secrets.RUBYGEMS_AUTH_TOKEN
        if: ${{ steps.release.outputs.release_created }}
```

### Log output

```text
https://github.com/coaxial/maid/actions/runs/4493366291/jobs/7904477828
```

### Additional information

https://guides.rubygems.org/patterns/#prerelease-gems
https://github.com/googleapis/release-please/blob/main/src/version.ts#L18

**Comment by coaxial** [comment]
See [googleapis/release-please/#1886](https://github.com/googleapis/release-please/issues/1886)

**Comment by chingor13** [comment]
This would need to be fixed on the upstream repo for which you've filed a bug.
